runtime/reconcilers/model.go

```diff
@@ -1547,6 +1562,59 @@ func (r *ModelReconciler) shouldTrigger(ctx context.Context, self *runtimev1.Res
 	}
 }
 
+// runModelTests executes the user defined tests for the model
+func (r *ModelReconciler) runModelTests(ctx context.Context, self *runtimev1.Resource) error {
+	if len(self.GetModel().Spec.Tests) == 0 {
+		r.C.Logger.Debug("No tests defined for model", zap.String("model", self.Meta.Name.Name), observability.ZapCtx(ctx))
+		return nil
+	}
+	var errs []error
+
+	for _, test := range self.GetModel().Spec.Tests {
+		result, err := r.C.Runtime.Resolve(ctx, &runtime.ResolveOptions{
```

Comment:
Add `Args: map[string]any{"limit": 1}` as well here, several resolvers support that to optimize to not return too many rows (since just one row already means the test failed). Example here: https://github.com/rilldata/rill/blob/644ee987a7961b84a23328b212a412a43e1f72b1/runtime/reconcilers/alert.go#L570